addons/payment_stripe/models/payment_provider.py

```diff
@@ -498,7 +498,11 @@ def _stripe_get_inline_form_values(
                     'postal_code': partner.zip or '',
                 },
             },
-            'is_tokenization_required': self._is_tokenization_required(**kwargs),
+            'is_tokenization_required': (
+                self.allow_tokenization
+                and (not payment_method_sudo or payment_method_sudo.support_tokenization)
+                and self._is_tokenization_required(**kwargs)
+            ),
```

Comment:
To keep consistent with other payment flows
```suggestion
            'is_tokenization_required': PaymentPortal._compute_show_tokenize_input_mapping(
                self
            )[self.id] and payment_method_sudo.support_tokenization,
```

Reply:
All payment methods available will be shown no, so it's up to the user to activate only tokenizable method or not.
But if tokenizable, the customer is force to tokenize.

Reply:
@chevalierv Thanks for the review, I've updated the code to use the helper (as you suggested) but with two changes:
- reversing the condition for the helper, if we're **not** showing the input it mean the tokenization is required
- passing `**kwargs`, otherwise we're missing [`sale_order_id`](<https://github.com/odoo/odoo/blob/18.0/addons/payment_stripe/views/payment_stripe_templates.xml#L12>) and the result  won't match what's done in `_create_transaction()`

Reply:
Hello @xavieralt 

I missed this but it looks like we can't use the helper in the way you want because we hide the checkbox if the tokenization is required, or if the tokenization is disabled on the provider itself. This prevents payment for all customers who have disabled the "Allow saving payment methods" on the provider.

```python
            'is_tokenization_required': self.allow_tokenization and self._is_tokenization_required(**kwargs) and payment_method_sudo.support_tokenization,
```

I'm reverting this PR for now in 18.0 and 18.1, can you fix the forward port for 18.2+ and a new PR for 18.0 and 18.1?

Reply:
Ok, so it means we're reverting to my initial proposition. I've updated the 18.2 PR https://github.com/odoo/odoo/pull/209658 if you want to double check ? :pray:

Reply:
Indeed, sorry that I mislead you 😅 I've quickly checked and it looks good :) I'll let you r+

Reply:
Indeed, sorry that I mislead you 😅 I've quickly checked and it looks good :)